doc(talos): v2 engine support (#981)
longhorn/website: content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md

For detailed instructions, see the Talos documentation on [Editing Machine Configuration](https://www.talos.dev/v1.6/talos-guides/configuration/editing-machine-configuration/).

> ## v2 Data Engine

Reviewer: ```suggestion
## V2 Data Engine
```

> To use the v2 data engine volumes, ensure your nodes fulfill the v2 data engine [prerequisites](../../../v2-data-engine/prerequisites#prerequisites).

Reviewer: ```suggestion
To use V2 volumes, all nodes must meet the V2 Data Engine [prerequisites](../../../v2-data-engine/prerequisites#prerequisites).
```

```yaml
machine:
  sysctls:
    vm.nr_hugepages: "1024"
  kernel:
    modules:
      - name: nvme_tcp
      - name: vfio_pci
#     - name: uio_pci_generic
```

> **Note:** Talos Linux version 1.7.x and prior versions do not include the `uio_pci_generic` kernel module. If your system device supports `vfio_pci`, it's the preferred method and you won't need to install and enable the `uio_pci_generic` kernel driver, for more details, see [SPDK: System Configuration User Guide](https://spdk.io/doc/system_configuration.html). If `vfio_pci` is incompatible with your system or specific hardware, you will need to use `uio_pci_generic`. Future version of Talos Linux are expected to include native support for `uio_pci_generic`, see GitHub issue - [uio_pci_generic can be required for DPDK or SPDK](https://github.com/siderolabs/talos/issues/9236).

Reviewer: ```suggestion
> **Note:** 
> Talos Linux v1.7.x and earlier versions do not include the `uio_pci_generic` kernel module. If your system device supports `vfio_pci`, which is the preferred kernel module for SPDK application deployment, you are not required to install and enable the `uio_pci_generic` kernel driver. For more information, see [System Configuration User Guide](https://spdk.io/doc/system_configuration.html) in the SPDK documentation.
>
> You can use `uio_pci_generic` if `vfio_pci` is incompatible with your system or specific hardware. Future versions of Talos Linux are expected to include native support for `uio_pci_generic`. For more information, see [Issue #9236](https://github.com/siderolabs/talos/issues/9236).
```
